egnoramoose.py

```diff
@@ -1,119 +1,167 @@
 class Pawn:
-	def __init__(self,location):
-		self.location = location
+    pass
+
+
+class Location:
+    def __init__(self, *location):
+        self.row = -1
+        self.column = -1
+        self._convert(location)
+
+    def __sub__(self, other):
+        return Location(self.row-other.row, self.column-other.column)
+
+    def __str__(self):
+        return '{}{}'.format(self.row, self.column)
+
+    @staticmethod
+    def _list_to_int(list_obj):
+        return [int(i) for i in list_obj]
+
+    def _convert(self, location):
```

Comment:
As stated above, :param `location` is now a tuple. If the tuple has size 1, this mean one parameter was given. The assert ensures that the one parameter has a length of 2, i.e '00', [1,2].

In the case of 2 arguments, i.e Location(2,3), Location('2','3'). The assert ensures each arg has a size of 1.

There are potential problems here, if the integer is multiple digits, different settings in the game will be thrown off as this dont check that